src/storage/src/render/persist_sink.rs

```diff
@@ -51,35 +50,33 @@ pub fn render<G>(
 
     let mut input = persist_op.new_input(&source_data.inner, Exchange::new(move |_| hashed_id));
 
-    let shared_frontier = Rc::clone(&storage_state.source_uppers[&src_id]);
+    let current_upper = Rc::clone(&storage_state.source_uppers[&src_id]);
 
     let weak_token = Rc::downgrade(&token);
 
     persist_op.build_async(
         scope.clone(),
         move |mut capabilities, frontiers, scheduler| async move {
             capabilities.clear();
-            let mut buffer = Vec::new();
-            let mut stash: HashMap<_, Vec<_>> = HashMap::new();
-
             let mut write = crate::persist_cache::open_writer::<
                 SourceData,
                 (),
                 mz_repr::Timestamp,
                 mz_repr::Diff,
-            >(
-                storage_metadata.persist_location,
-                storage_metadata.persist_shard,
-            )
+            >(metadata.persist_location, metadata.data_shard)
             .await
             .expect("could not open persist shard");
 
+            *current_upper.borrow_mut() = write.upper().clone();
```

Comment:
can you add a comment explaining what this does?

Reply:
something like:
```
// initialize this sinks `upper` to the `upper` of the persist shard we are writing do. Data from the source before this time will be dropped, as it has already been persisted
// In the future, sources will avoid passing through data before this upper, see <link to issue>

Reply:
Thanks, added a comment